x-pack/plugins/fleet/common/services/output_helpers.test.ts

```diff
@@ -45,3 +48,55 @@ describe('getAllowedOutputTypeForPolicy', () => {
     expect(res).toEqual(['elasticsearch']);
   });
 });
+
+describe('outputYmlIncludesReservedPerformanceKey', () => {
+  describe('dot notation', () => {
+    it('returns true when reserved key is present', () => {
+      const configYml = `queue.mem.events: 1000`;
+
+      expect(outputYmlIncludesReservedPerformanceKey(configYml)).toBe(true);
+    });
+
+    it('returns false when no reserved key is present', () => {
+      const configYml = `some.random.key: 1000`;
+
+      expect(outputYmlIncludesReservedPerformanceKey(configYml)).toBe(false);
+    });
+  });
+
+  describe('object notation', () => {
+    it('returns true when reserved key is present', () => {
+      const configYml = `
+      queue:
+          mem:
+            events: 1000
+    `;
+
+      expect(outputYmlIncludesReservedPerformanceKey(configYml)).toBe(true);
+    });
+
+    it('returns false when no reserved key is present', () => {
+      const configYml = `
+        some:
+          random:
+            key: 1000
+      `;
+
+      expect(outputYmlIncludesReservedPerformanceKey(configYml)).toBe(false);
+    });
+  });
+
+  describe('plain string', () => {
+    it('returns false when no reserved key is present', () => {
+      const configYml = `bulk_max_size`;
+
+      expect(outputYmlIncludesReservedPerformanceKey(configYml)).toBe(true);
+    });
+
+    it('returns true when reserved key is present', () => {
+      const configYml = `just a string`;
+
+      expect(outputYmlIncludesReservedPerformanceKey(configYml)).toBe(false);
+    });
+  });
+});
```

Comment:
Relevant YML parsing test cases here - we also still check plain strings even if the YML is invalid just to be sure.